lib/datahub-client/data_platform_catalogue/client/graphql_helpers.py

```diff
@@ -293,3 +296,19 @@ def parse_relations(
 
     relations_return = {relationship_type: related_entities}
     return relations_return
+
+
+def _make_user_email_from_urn(urn) -> str:
+    """
+    Creates a user email using a user entity urn. This should only be called
+    when an urn exists for a user that has not signed into datahub via sso,
+    so has not been created as an entity and has no associated email address.
+
+    We will look to revist our approach to ownership user creation, see
+    github issue https://github.com/ministryofjustice/find-moj-data/issues/578,
+    but for now this will fix the issue of owners being flagged in datahub but not
+    showing in find-moj-data
+    """
+    email_parts = urn.replace("urn:li:corpuser:", "").split(".")
+    email = f"{email_parts[0]}.{email_parts[1]}@justice.gov.uk"
```

Comment:
would this work? 

```suggestion
  username = urn.replace("urn:li:corpuser:", "")
  email = f"{username}@justice.gov.uk"
```

I'm not sure we should be relying on emails always having two names separated by a dot. I haven't checked in entra but seems possible that someone could have an email with 2 dots in, or none.

Could you add unit tests for this as well?

Reply:
This will also break if we get any users that can log into entra but don't have a justice.gov.uk email. But I think it's OK as a workaround for now. Just something to consider when we pick up that other ticket.